### DISCUSSION TITLE
How to update contents of, and then refresh, ScrollView?

### DISCUSSION THREAD

**Discussion by jeffwright13** [Textualize/textual#417] [Q&A] [answered]
Let's say I have a ScrollView instance, named `body`. How do I update its contents? Is it as simple as `body = Text(<new_content>)`?

And assuming I was successful in rewriting my ScrollView's content, how do I get it to refresh on screen? Would it be `body.refresh()`?

**Comment by HiPeople21** [comment, marked as answer]
ScrollViews have an `update` method that takes two arguments: the `renderable` and `home`, which is a boolean that is defaulted to true. If it is true, it moves the ScrollView back to the top.

**Comment by jeffwright13** [comment]
Thanks for the tip. I was able to invoke the update() method, but I don't see anything changing in the TUI. Do I need to do a refresh manually?

I did notice that I see a `RuntimeWarning: coroutine 'ScrollView.update' was never awaited` message. Not being all that comfortable with async, I'm not really sure where to put this `await`. That is, assuming I need to put it in myself at all. I would have expected the `update` method to just be callable, without me having to worry about an `await` somewhere, given that the `update` method's definition includes its own `await`:

```
async def update(self, renderable: RenderableType, home: bool = True) -> None:
    if home:
        self.home()
    await self.window.update(renderable)
```

**Reply by jeffwright13** [reply]
It turns out that you just pre-pend the call to `update` with `await`, like is shown in the example file "big_table.py". Case closed.
